### PR TITLE
removed build-time condition SDK >= 10.14

### PR DESCRIPTION
**Pull request by tijsvbricsys**
"trigger redraw upon shown for layer-backed views" is also needed when building with sdk 10.13

the build-time check that the used SDK must be 10.14 or higher is not needed
I checked building with SDK 10.12 and 10.13

the used NsView functions are available in far older SDK's
* https://developer.apple.com/documentation/appkit/nsview/1483475-setneedsdisplay
 Availability macOS 10.0+ 
* https://developer.apple.com/documentation/appkit/nsview/1483473-ishiddenorhashiddenancestor
 Availability macOS 10.3+

I did not add an ifdef checking for sdk 10.3 or higher - I assume the minimum requirement for building wxWidgets is higher than 10.3

see also https://github.com/wxWidgets/wxWidgets/commit/e484a2db1926374311ec4327605e8935d04ac873

### CODE REVIEW

**Comment by tijsvbricsys** [comment]
@csomor @artur-so

I could build against SDK's 10.13 and 10.12.

I could not reproduce in a sample.  Perhaps it is not required for this build-time only change, but it would still be nice to be able to reproduce...

In our sources (I am a colleague of Artur) the issue occurs in a wxMiniFrame with wxCustomBackgroundWindow<wxWindow> children.  Refresh on the wxMiniFrame does not Refresh the children (onPaint of wxCustomBackgroundWindow is not triggered).
I adapted the wxMiniFrame in the dialogs sample to mimick the situation, but I must be missing a necessary condition.
One of the comments mentions "Layer-backed views".  Any hints on how to make a view layer-backed?

**Comment by csomor** [comment]
Thanks @tijsvbricsys for trying to isolate things. 
The dependency on SDK 10.14 is described here https://developer.apple.com/documentation/macos-release-notes/appkit-release-notes-for-macos-10_14, 
I'm trying to find out what could trigger this on earlier systems. I'm only calling setWantsLayer on the wxAVView... I'm not aware of another method that could lead to such behaviour.

**Comment by vadz** [comment]
BTW, is this the same as https://trac.wxwidgets.org/ticket/19165 or something else?

**Comment by tijsvbricsys** [comment]
This is the same as https://trac.wxwidgets.org/ticket/19165

I meanwhile further attempted to reproduce by mimicking the problematic window from our application in the dialogs sample, but I failed to reproduce so far, even if I build the same sample with our customized wxwidgets binaries.

In our application we have added the property [self setWantsLayer:YES]; to glcanvas.mm.
I suspect that it influences the toplevel frame of our application.  According to the [wantsLayer documentation](https://developer.apple.com/documentation/appkit/nsview/1483695-wantslayer) it is inherited by children:

> Creating a layer-backed view implicitly causes the entire view hierarchy under that view to become layer-backed.

I suspect the presence of the customized glcanvas causes the issue in our application.
The next thing I had in mind was to fiddle with setWantsLayer a bit to check that.

I agree with @afalkenhahn in https://trac.wxwidgets.org/ticket/19165 that the compile-time condition doesn't make sense, since the runtime check is available in earlier sdks.

I think the runtime check for running on 10.14 is fine.  Perhaps checking the wantsLayer property would be more correct, but not necessarily better in practice.

**Comment by tijsvbricsys** [comment]
I locally removed [self setWantsLayer:YES]; from our customized glcanvas.mm, but the refresh problem still reproduces in our application (built against 10.13 sdk, run on macOS 11.1).  Sorry, I can't seem to pinpoint what triggers the issue...

**Comment by vadz** [comment]
Should this be still merged? It also makes sense to me, but (my) common sense and Mac are not always compatible...

**Comment by tijsvbricsys** [comment]
tl;dr I have no strong opinion either way.

>(my) common sense and Mac are not always compatible...
Same for me I'm afraid.

To build for minimum requirement 10.13 it makes most sense to me to actually build using SDK 10.13.
However, it is also possible to use a newer SDK and set the 'deployment target' to 10.13.
My impression is that Apple assumes everybody uses the newest or second-newest SDK, therefore their documentation does not cover our setup.

For us (as in: my company) it is not important to merge this change.  As it happens, I am upgrading our internal builds to using SDK 10.15.  (for a micro update of a 10.13 based version we applied the change in a release branch, for our development branch it no longer matters)

Since the actual trigger could not be pinpointed I have no strong opinion on what would be best - myself I tend to prefer doing nothing in such case, so I totally understand if it is not merged.

**Comment by csomor** [comment]
Since we also have the report from Andreas, I'll merge this.

**Comment by csomor** [comment]
@tijsvbricsys Thanks Tijs :-)
